src/plugins/vis_builder/public/application/components/data_tab/secondary_panel.tsx

```diff
@@ -98,49 +103,98 @@ export function SecondaryPanel() {
-              );
-
-              if (indexField) {
-                aggConfig.params.field = indexField;
-              }
-            }
-
-            dispatch(editDraftAgg(aggConfig.serialize()));
-          }}
-        />
+        >
+          <EditorErrorBoundary>
+            <DefaultEditorAggParams
+              className="vbConfig__aggEditor"
+              agg={aggConfig!}
+              indexPattern={indexPattern!}
+              setValidity={handleSetValid}
+              setTouched={setTouched}
+              schemas={schemas}
+              formIsTouched={touched}
+              groupName={selectedSchema?.group ?? 'none'}
+              metricAggs={metricAggs}
+              state={{
+                data: {},
+                description: '',
+                title: '',
+              }}
+              setAggParamValue={function <T extends string | number | symbol>(
+                aggId: string,
+                paramName: T,
+                value: any
+              ): void {
+                aggConfig.params[paramName] = value;
+                dispatch(editDraftAgg(aggConfig.serialize()));
+              }}
+              onAggTypeChange={function (aggId: string, aggType: IAggType): void {
+                aggConfig.type = aggType;
+
+                // Persist field if the new agg type supports the existing field
+                const fieldParam = (aggType.params as AggParam[]).find(
+                  ({ type }) => type === 'field'
+                );
+                if (fieldParam) {
+                  const availableFields = (fieldParam as IFieldParamType).getAvailableFields(
+                    aggConfig
+                  );
+                  const indexField = availableFields.find(
+                    ({ name }) => name === get(draftAgg, 'params.field')
+                  );
+
+                  if (indexField) {
+                    aggConfig.params.field = indexField;
+                  }
+                }
+
+                dispatch(editDraftAgg(aggConfig.serialize()));
+              }}
+            />
+          </EditorErrorBoundary>
+        </OpenSearchDashboardsContextProvider>
       )}
     </div>
   );
 }
+
+class EditorErrorBoundary extends Component<{}, { error?: any }> {
+  state = {
+    error: undefined,
+  };
+
+  static getDerivedStateFromError(error: any) {
+    return { error };
+  }
+
+  componentDidCatch(error) {
+    // eslint-disable-next-line no-console
+    console.error(error);
+  }
+
+  render() {
+    if (this.state.error) {
+      return (
+        <EuiCallOut
+          title={i18n.translate('visBuilder.aggParamsEditor.errorTitle', {
+            defaultMessage: 'Error',
+          })}
+          color="danger"
+          iconType="alert"
+        >
+          <p>
+            {i18n.translate('visBuilder.aggParamsEditor.errorMsg', {
+              defaultMessage: 'Something went wrong while editing the aggregation',
+            })}
+          </p>
+        </EuiCallOut>
```

Comment:
Does this need UX signoff? Why not use a toast as we are for other errors?

Reply:
The toast error pattern is not ideal in the first place because even though the error persists, the toast will disappear. Also this is an error boundary that the user should no longer run into. Its just a safety mechanism to make sure that if it ever does, the whole app does not crash, since we inherit the `DefaultAggParams` component from visualize. 

Also checking with @KrooshalUX about UX signoff.

Reply:
@KrooshalUX is okay with this UX but would also like a toast message to be shown as a part of the error. However that is not a blocker for this and I've created an [issue](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3231) to discuss the use of toasts in error boundaries.